syntax = "proto3";

package boltz;

import "common.proto";

// Inference service for protein structure prediction
service InferenceService {
  // Predict protein structure from sequence
  rpc PredictStructure (PredictionRequest) returns (PredictionResponse) {}
  
  // Get prediction job status
  rpc GetJobStatus (JobStatusRequest) returns (JobStatusResponse) {}
  
  // Cancel prediction job
  rpc CancelJob (CancelJobRequest) returns (CancelJobResponse) {}
}

// Request for structure prediction
message PredictionRequest {
  string job_id = 1;
  string sequence = 2;
  int32 recycling_steps = 3;
  int32 sampling_steps = 4;
  int32 diffusion_samples = 5;
  string output_format = 6;
  string model_version = 7;
}

// Response for structure prediction
message PredictionResponse {
  string job_id = 1;
  string status = 2;
  string result_path = 3;
  string error_message = 4;
}
